File: examples/robot/ExampleRunTaskAndGetPrintResult.cs
using System;
using lebai;

namespace lebai.app
{
    class ExampleRunTaskAndGetPrintResult
    {
        static int Main(string[] args)
        {
            if (args.Length < 2 || args.Length > 3)
            {
                Console.Error.WriteLine(
                    $"Usage: {AppDomain.CurrentDomain.FriendlyName} <robot_ip> <task_name> [sim]");
                return 1;
            }

            if (args.Length == 3 && args[2] != "sim")
            {
                Console.Error.WriteLine("The optional third argument must be 'sim'");
                return 1;
            }

            using (var robot = new Robot(args[0], args.Length == 3))
            {
                robot.StartSys();

                uint taskId = robot.Native.start_task(args[1]);
                Console.WriteLine($"Started task {taskId}");

                string printResult = robot.Native.wait_task(taskId);
                Console.WriteLine("Task print result:");
                Console.Write(printResult);
                if (printResult.Length == 0 || !printResult.EndsWith("\n", StringComparison.Ordinal))
                {
                    Console.WriteLine();
                }
            }

            return 0;
        }
    }
}

File: examples/robot/ExampleRunTaskAndGetPrintResult.java
package org.sdk.lebai.robot;

import org.sdk.lebai.Robot;

public final class ExampleRunTaskAndGetPrintResult {
  public static void main(String[] args) {
    if (args.length < 2 || args.length > 3) {
      System.err.println(
          "Usage: ExampleRunTaskAndGetPrintResult <robot_ip> <task_name> [sim]");
      return;
    }

    if (args.length == 3 && !"sim".equals(args[2])) {
      System.err.println("The optional third argument must be 'sim'");
      return;
    }

    try (Robot robot = new Robot(args[0], args.length == 3)) {
      robot.startSys();

      long taskId = robot.getNative().start_task(args[1]);
      System.out.println("Started task " + taskId);

      String printResult = robot.getNative().wait_task(taskId);
      System.out.println("Task print result:");
      System.out.print(printResult);
      if (printResult.isEmpty() || !printResult.endsWith("\n")) {
        System.out.println();
      }
    }
  }

  private ExampleRunTaskAndGetPrintResult() {}
}
